docs/doc/60-contributing/03-rfcs/20220909-avoid-duplicate-when-copy-into-table.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Avoid duplicate when copy stage files into a table
+description: Avoid duplicating when copy stage files into a table
+---
+
+- Tracking Issue: <https://github.com/datafuselabs/databend/issues/6338>
+
+## Summary
+
+When streaming copy stage files from a directory into a table, there is a chance that some files have already been copied, So it needs some ways to avoid duplicate copying files.
+
+## Save copy into table stage files meta information in meta service
+
+Whenever copy stage files into a table, save the stage file meta information into the meta service:
```

Comment:
is the stage files metadata updated in a transaction with the data commit?

Reply:
> is the stage files metadata updated in a transaction with the data commit?

what do you mean about `data commit`, i guess it means `copy the stage files into table`, if this is what you means, the answer is no, cause `copy the stage files into table` happen in object store, but `table stage files metadata updated` happen in meta service, they cannot be in a transaction.